marshmallow_dataclass/__init__.py

```diff
@@ -303,7 +331,9 @@ def class_schema(
     )
 
     schema_class = type(clazz.__name__, (_base_schema(clazz, base_schema),), attributes)
-    return cast(Type[marshmallow.Schema], schema_class)
+    result = cast(Type[marshmallow.Schema], schema_class)
+    SCHEMA_REGISTRY[(clazz, base_schema)] = result
```

Comment:
Your cache can only grow. This is a memory leak.

Reply:
I was going to add a max-size but then decided against it as that would re-introduce the duplicate issue, just at some arbitrary point further down. E.g. after 128 generated schemas.

Furthermore, it's not a cache per se, it's more of a registry. It'll grow to equal the size of unique dataclass/schemas and then stop growing.

Reply:
There is no guarantee that the number of unique schemas is bounded. I agree that this means we cannot **guarantee** that we will always return the same schema, but I think that's fine, or at least better than a memory leak.

Reply:
Okay sure, fair point. What number would you suggest? 1024, perhaps? Default is 128 in lru_cache, which might cause issues more the general use-case.

Reply:
Yes, 1024 seems reasonable 👍